firedrake/tsfc_interface.py

```diff
@@ -315,14 +340,10 @@ def extract_numbered_coefficients(expr, numbers):
     return coefficients
 
 
-def _make_interface_key(interface, form):
-    if interface:
-        # The 'interface' argument is a small hack done in patch.py. When
-        # specified, what really matters for caching is which coeffients
-        # are passed to the 'dont_split' kwarg.
-        assert isinstance(interface, functools.partial)
-        assert interface.func is KernelBuilder
-        coefficients = form.coefficients()
-        return tuple(coefficients.index(f) for f in interface.keywords["dont_split"])
-    else:
-        return None
+def _make_dont_split_numbers(dont_split, form):
+    temp = tuple(
+        form.coefficients().index(c)
+        for c in dont_split
+        if c in form.coefficients()
+    )
+    return tuple(sorted(temp))
```

Comment:
```suggestion
    return tuple(
        sorted(
            form.coefficients().index(c)
            for c in dont_split
            if c in form.coefficients()
        )
    )
```
No need to allocate a temporary tuple.